src/order_executer/service/portfolio.py

```diff
@@ -1,28 +1,114 @@
+                if amount >= min_amount:
+                    portfolio.buy(currency, amount, data)
             elif prediction == -1:
-                portfolio.sell(asset)
+                amount = portfolio.get_available_amount(currency=currency, buying=False, aggressiveness=0.10, data=data)
+                if amount >= min_amount:
+                    portfolio.sell(currency, amount, data)
+            else:
+                print("HODL!")
+
+    def update_wallet_worth(self, data):
+        for portfolio in self.portfolios:
+            portfolio.update_wallet_worth(data)
 
 
 class Portfolio:
-    def __init__(self, owner_name, binance_key, binance_pass):
+    def __init__(self, owner_name, client=BinanceClient, api_key=None, api_secret=None):
         self.owner_name = owner_name
-        self.binance_key = binance_key
-        self.binance_pass = binance_pass
-        self.assets = {}
+        self.client = client(api_key=api_key, api_secret=api_secret)
+        self.wallet_worth_history = []
+
+    def buy(self, currency, amount, data):
+        """
+        Currently in test mode, sends and validates a request to Binance API but does not really trade.
+        """
+        print(f"Buying {amount} {currency}")
+        self.client.trade(
+            currency_to_buy=currency,
+            currency_to_sell=config.CURRENCY_TO_SELL,
+            action=enums.SIDE_BUY,
+            amount=amount,
+            data=data
+        )
+
+    def sell(self, currency, amount, data):
+        """
+        Currently in test mode, sends and validates a request to Binance API but does not really trade.
+        """
+        print(f"Selling {amount} {currency}")
+        self.client.trade(
+            currency_to_buy=currency,
+            currency_to_sell=config.CURRENCY_TO_SELL,
+            action=enums.SIDE_SELL,
+            amount=amount,
+            data=data
+        )
+
+    def update_wallet_worth(self, data):
+        total_balance = 0
+        balances = self.client.get_balances()
+        for currency, balance in balances.iterrows():
+            balance = balance['free']
+            if not currency == config.CURRENCY_TO_SELL:
+                balance = self.convert(balance, currency, config.CURRENCY_TO_SELL, data)
+            total_balance += balance
+        self.wallet_worth_history.append(total_balance)
+
+    def get_available_amount(self, currency, buying, aggressiveness, data):
+        """
+        Get available crypto amount to trade. It depends on available funds,
+        the level of aggressiveness, and the maximum exchange precision.
+
+        :param currency: Crypto to buy/sell
+        :param buying: If true, checks the amount of the currency to sell, otherwise, currency to buy
+        :param aggressiveness: Percent of available funds to trade
+        :return: Available amount to trade (in crypto)
+        """
+        if buying:
```

Comment:
Why do we need this clause? Shouldn't get_available_amount just return the available cash?

Reply:
Yes, but the available cash in *crypto*. Meaning, if you buy BTC with USDT, you do: `trade('BTCUSDT', action='BUY'). If you sell BTC to USDT, then you do: `trade('BTCUSDT', action='SELL'). But in both cases you need to specify the amount in crypto. 

The reason is, there is no currency pair like `USDTBTC` or `EURBTC`, so I had to find this workaround